haicrypt/hcrypt_sa.c

```diff
@@ -37,7 +37,7 @@ int hcryptCtx_SetSecret(hcrypt_Session *crypto, hcrypt_Ctx *ctx, const HaiCrypt_
 		ctx->cfg.pwd_len = 0;
 		/* KEK: Key Encrypting Key */
 		if (0 > (iret = crypto->cryspr->km_setkey(crypto->cryspr_cb,
-							  (HCRYPT_CTX_F_ENCRYPT & ctx->flags ? true : false),
+							  ((HCRYPT_CTX_F_ENCRYPT & ctx->flags) ? true : false),
```

Comment:
Hopefully this will shut-up the code analyser. There is 0-ambiguity with the "C" pririty of operators in the case.